core/ledger/pvtdatastorage/store.go

```diff
@@ -831,6 +835,36 @@ func (s *Store) purgeExpiredData(minBlkNum, maxBlkNum uint64) error {
 	return nil
 }
 
+func (s *Store) deleteDataMarkedForPurge() error {
+	maxBatchSize := 4 * 1024 * 1024 // 4Mb
+	p := newPurgeUpdatesProcessor(s.db, maxBatchSize)
+	pStart, pEnd := rangeScanKeysForPurgeMarkers()
```

Comment:
Since PurgeMarkers are never deleted, does this imply that we'll be iterating over all previously purged keys as well as the newly purged keys every time? With the implication that it will take longer and longer as time goes on.

Reply:
There will be at most one purge marker for a particular key. So, if a new purge operation is issued, the existing purge marker is over-written. So, if with time the purging gets issued for new keys, yes the scanning would be longer. But since, the hashed index keys would be purged, the processing would stop at scanning this list.

There are two ways to avoid this scanning as well - 1) Maintain another copy of the purge marker for this purpose and keep deleting from there. Or some complex garbage collection for the purge markers if and when there is no corresponding missing data. As of now, I am not sure if we would want to go to that level of optimization for this, as this is a background operation and performs a range scan. But certainly these options are there if we need them in future.

Reply:
So if 1 million keys have ever been purged, every time we attempt to purge we will iterate through all million records, and for every one of them query for corresponding HashedIndex keys (one million range queries), even if there is no work to be done?

We should at least open an issue to identity this eventual need for optimization.